File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/input/model/DolphinVibratorManager.java
// SPDX-License-Identifier: GPL-2.0-or-later

package org.dolphinemu.dolphinemu.features.input.model;

import android.os.Vibrator;

import androidx.annotation.Keep;
import androidx.annotation.NonNull;

/**
 * A wrapper around {@link android.os.VibratorManager}, for backwards compatibility.
 */
public interface DolphinVibratorManager
{
  @Keep @NonNull
  Vibrator getVibrator(int vibratorId);

  @Keep @NonNull
  int[] getVibratorIds();
}

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/input/model/DolphinVibratorManager.kt
// SPDX-License-Identifier: GPL-2.0-or-later

package org.dolphinemu.dolphinemu.features.input.model

import android.os.Vibrator
import androidx.annotation.Keep

/**
 * A wrapper around [android.os.VibratorManager], for backwards compatibility.
 */
@Keep
interface DolphinVibratorManager {
    fun getVibrator(vibratorId: Int): Vibrator

    fun getVibratorIds(): IntArray

    fun getDefaultVibrator(): Vibrator
}
